src/BattleServer/battle.cpp

```diff
@@ -4442,3 +4442,21 @@ bool BattleSituation::preTransPoke(int s, Pokemon::uniqueId check)
     }
     return false;
 }
+
+bool BattleSituation::canMegaEvolve (int slot) {
+    if (megas[player(slot)]) {
+        return false;
+    }
+    if (ItemInfo::isMegaStone(poke(slot).item())) {
+        if (ItemInfo::MegaStoneForme(poke(slot).item()).original() == poke(slot).num()) {
+            return true;
+        }
+        if (ItemInfo::MegaStoneForme(poke(slot).item()).original() == Pokemon::uniqueId(poke(slot).num().pokenum,0)) {
```

Comment:
Thinking about it, there's no reason to not just replace the first if statement with this. There's no situation in normal play where a Mega Evo can hold a Mega Stone and still be able to avoid, so it shouldn't be an issue. There's currently no formes that can mega evolve either (although if they could, I'd imagine they use the base forme mega evos anyway)